plugins/module_utils/redfish_utils.py

```diff
@@ -2792,3 +2792,81 @@ def set_manager_nic(self, nic_addr, nic_config):
         if response['ret'] is False:
             return response
         return {'ret': True, 'changed': True, 'msg': "Modified Manager NIC"}
+
+    def set_hostinterface_attributes(self, hostinterface_config):
+        response = self.get_request(self.root_uri + self.manager_uri)
+        if response['ret'] is False:
+            return response
+        data = response['data']
+        if 'HostInterfaces' not in data:
+            return {'ret': False, 'msg': "HostInterfaces resource not found"}
+
+        hostinterfaces_uri = data["HostInterfaces"]["@odata.id"]
+        response = self.get_request(self.root_uri + hostinterfaces_uri)
+        if response['ret'] is False:
+            return response
+        data = response['data']
+        uris = [a.get('@odata.id') for a in data.get('Members', []) if
+                a.get('@odata.id')]
+
+        # Fail if multiple Redfish Host Interfaces are detected
+        if len(uris) != 1:
```

Comment:
In the vast majority of cases, there should only be one host interface on a manager, but there are designs we want to allow where one manager is managing multiple systems, thus allowing for multiple host interfaces. I think it would be good to have the option to allow a user to specify a host interface instance; if not provided, then simply go for the only interface.

Essentially:

```
if host_interface_id is specified and member exists:
    # Use exact interface
elif len(host_interface_list) == 1:
    # Use single host interface
else:
    # No matching interface; error
```